### PR TITLE
apprt/gtk: fix the combination of gtk-titlebar=false and gtk-tabs-location=hidden

### LINKED ISSUE #3178: Ghostty breaks when setting `gtk-titlebar = false`, and `gtk-tabs-location = hidden` on Linux
Hi, i'm trying to configure tabs because i don't really want to use tmux, i just want to remove visualy the tabs completely, because they take space. My issue is when setting `gtk-tabs-location = hidden`. Ghostty breaks to one single color like this:

Image

I'm running Arch Linux with Hyprland, installed Ghostty by building it from source with zig.

### PR DESCRIPTION
**Pull request by tristan957**
Fixes: #3178

### CODE REVIEW

File: src/apprt/gtk/Window.zig [hunk 1]
```diff
@@ -167,7 +167,7 @@ pub fn init(self: *Window, app: *App) !void {
         // If we're using an AdwWindow then we can support the tab overview.
         if (self.tab_overview) |tab_overview| {
             if (comptime !adwaita.versionAtLeast(1, 4, 0)) unreachable;
-            assert(self.isAdwWindow());
+            assert(self.app.config.@"gtk-adwaita" and adwaita.versionAtLeast(1, 3, 0));
```
Comment (jcollie) [inline src/apprt/gtk/Window.zig:170]: Is there a reason that the version numbers are different?

Comment (tristan957) [reply src/apprt/gtk/Window.zig]: Fixed

File: src/apprt/gtk/Window.zig [hunk 2]
```diff
@@ -125,7 +125,7 @@ pub fn init(self: *Window, app: *App) !void {
     self.notebook = Notebook.create(self);
 
     // If we are using Adwaita, then we can support the tab overview.
-    self.tab_overview = if ((comptime adwaita.versionAtLeast(1, 3, 0)) and adwaita.enabled(&self.app.config) and adwaita.versionAtLeast(1, 3, 0)) overview: {
+    self.tab_overview = if ((comptime adwaita.versionAtLeast(1, 4, 0)) and adwaita.enabled(&self.app.config) and adwaita.versionAtLeast(1, 3, 0)) overview: {
```
Comment (jcollie) [inline src/apprt/gtk/Window.zig:128]: There's still checks for two different versions of Adwaita - comptime 1.4.0 and runtime 1.3.0.

**Review by jcollie — APPROVED** [review]

**Comment by adriandelgado** [comment]
This PR seems to break ghostty on some Linux instalations. Please see https://github.com/ghostty-org/ghostty/discussions/4097

**Comment by ttys3** [comment]
> This PR seems to break ghostty on some Linux instalations. Please see #4097

it also break the exists headbar, when setting gtk-titlebar = false and gtk-adwaita = true, no headbar is shown

https://github.com/ghostty-org/ghostty/discussions/4101

**Comment by tristan957** [comment]
> it also break the exists headbar, when setting gtk-titlebar = false and gtk-adwaita = true, no headbar is shown

This is expected.

**Pull request merged**